### PR TITLE
feat(eif-builder): add PCR0-covering signature section

### PR DESCRIPTION
**Pull request by jmt-lab**
Extend eif-builder library with EifSectionSignature (0x04) covering PCR0
via COSE_Sign1 ECDSA (ES256/ES384). Introduces a Signer trait with two
implementations:

* LocalSigner (aws-lc-rs) — reads a PEM ECDSA key from disk.
* KmsSigner (aws-sdk-kms) — calls KMS Sign on P-256/P-384 keys with
  typed errors for KmsAccessDenied / KmsKeyDisabled / KmsInvalidKeyState.

Wire format matches aws-nitro-enclaves-image-format: CBOR Array(1) of
Map { signing_certificate: PEM, signature: COSE_Sign1 } with a PcrInfo
payload for PCR0 = SHA-384(48 zero bytes || SHA-384(kernel || cmdline
|| ramdisks)). This makes `nitro-cli describe-eif` populate PCR8.

Also adds byte-preserving resign_eif() and describe_eif() library
entry points for the CLI subcommands landing in the next commit.

New deps: aws-lc-rs, coset, ciborium, pem, x509-cert.

Tests: PCR0 spec conformance, COSE payload shape, end-to-end signature
verification, DER parser edge cases (SEQUENCE length + trailing byte
rejection), resign round-trips.

### CODE REVIEW

**Review by Sparksssj — APPROVED** [review]

File: tools/eif-builder/src/signer.rs [hunk 1]
```diff
@@ -0,0 +1,673 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! EIF signature production for the `EifSectionSignature` (0x04) section.
+//!
+//! [`LocalSigner`] signs in-process from a PEM ECDSA key.
+//! [`KmsSigner`] delegates to AWS KMS. Both emit raw ECDSA `r || s`
+//! (RFC 8152 §8.1); P-256 → ES256, P-384 → ES384; other curves rejected.
```
Comment (ginglis13) [inline tools/eif-builder/src/signer.rs]: Can you link to the actual RFC if you're going to mention it? This particular RFC is obsoleted: https://www.rfc-editor.org/info/rfc8152/

File: tools/eif-builder/src/signer.rs [hunk 2]
```diff
@@ -0,0 +1,673 @@
...
+    /// COSE algorithm to advertise.
+    fn algorithm(&self) -> SignAlg;
+}
+
+/// In-process ECDSA signer backed by `aws-lc-rs`.
+pub struct LocalSigner {
+    key_pair: EcdsaKeyPair,
+    cert_pem: Vec<u8>,
+    #[cfg(test)]
+    public_key_spki_der: Vec<u8>,
```
Comment (ginglis13) [inline tools/eif-builder/src/signer.rs]: odd to have test only field in the pub struct?

File: tools/eif-builder/src/signer.rs [hunk 3]
```diff
@@ -0,0 +1,673 @@
...
+        if key_curve != cert_alg {
+            return CurveMismatchSnafu {
+                cert_curve: name_of(cert_alg),
+                key_curve: name_of(key_curve),
+            }
+            .fail();
+        }
+
+        #[cfg(test)]
+        let public_key_spki_der = spki_der_for_ecdsa(&raw_pub, key_curve);
```
Comment (ginglis13) [inline tools/eif-builder/src/signer.rs]: again why are we embedding test only data on the actual pub struct?

Can we just have a 

```rust
/// SPKI DER for the signing key. Test-only; derived on demand.
#[cfg(test)]
pub fn public_key_der(&self) -> Vec<u8> {
      spki_der_for_ecdsa(self.key_pair.public_key().as_ref(), self.alg)
}
```

for use in lib.rs?

File: tools/eif-builder/src/signer.rs [hunk 4]
```diff
@@ -0,0 +1,673 @@
...
+        if parsed_cert.tag() != "CERTIFICATE" {
+            return CertPemLabelSnafu {
+                label: parsed_cert.tag().to_string(),
+            }
+            .fail();
+        }
+        let alg = detect_cert_curve(parsed_cert.contents())?;
+
+        // Current-thread runtime so sync callers can invoke the async KMS SDK.
+        let runtime = tokio::runtime::Builder::new_current_thread()
```
Comment (ginglis13) [inline tools/eif-builder/src/signer.rs]: you could panic if already inside a tokio runtime, via either `block_on` or dropping the runtime. You could gate this with a new snafu and an early check in this method:

```rust
// This type owns a current-thread runtime and drives KMS via block_on.
// Both block_on and dropping the runtime panic if we're already inside a
// runtime, so refuse up front and turn a late panic into a typed error.
if tokio::runtime::Handle::try_current().is_ok() {
    return KmsInAsyncContextSnafu.fail();
}
```

File: tools/eif-builder/src/signer.rs (hunk 2 above)
Comment (jmt-lab) [reply tools/eif-builder/src/signer.rs]: good catch this is really weird.

File: tools/eif-builder/src/signer.rs (hunk 4 above)
Comment (jmt-lab) [reply tools/eif-builder/src/signer.rs]: I'm also questioning why this method isn't async to start with,

File: deny.toml [hunk 5]
```diff
@@ -54,14 +55,24 @@ license-files = [
     { path = "LICENSE", hash = 0x001c7e6c },
 ]
 
+[[licenses.clarify]]
+name = "half:1.8.3"
```
Comment (ginglis13) [inline deny.toml]: Is this still needed? Now that we've got rid of `serde_cbor`

File: clarify.toml [hunk 6]
```diff
@@ -5,6 +5,14 @@ ignore-licenses = [
     "Pixar"
 ]
 
+[clarify.half]
```
Comment (ginglis13) [inline clarify.toml]: shouldn't be needed anymore

File: Cargo.toml [hunk 7]
```diff
@@ -113,8 +114,10 @@ bit_field = "0.10"
 bon = "3"
 bounded-integer = "0.5"
 bytes = "1"
+ciborium = "0.2"
```
Comment (ginglis13) [inline Cargo.toml]: this is now only a transitive dep
```suggestion
```

File: deny.toml [hunk 8]
```diff
@@ -54,14 +55,24 @@ license-files = [
...
+
 [bans]
 # Deny multiple versions or wildcard dependencies.
 multiple-versions = "deny"
 wildcards = "deny"
 
 skip = [
     # several dependencies are using multiple versions of base64
-    { name = "base64" },
+    { name = "half" },
```
Comment (ginglis13) [inline deny.toml]: skip no longer needed, this is a no-op

File: tools/eif-builder/src/lib.rs [hunk 9]
```diff
@@ -286,26 +320,556 @@ pub fn build_eif(
...
+///   "sections": [
+///     { "type": "KERNEL",   "type_id": 1, "size": 50331648 },
+///     { "type": "CMDLINE",  "type_id": 2, "size":      431 },
+///     { "type": "METADATA", "type_id": 5, "size":      254 },
+///     { "type": "SIGNATURE","type_id": 4, "size":     2115 }
+///   ],
+///   "metadata": { "ImageName": "...", "BuildMetadata": { ... } }
+/// }
+/// ```
+///
```
Comment (ginglis13) [inline tools/eif-builder/src/lib.rs]:
```suggestion
/// Summarize an EIF's contents as JSON.
```

File: tools/eif-builder/src/signer.rs [hunk 10]
```diff
@@ -0,0 +1,668 @@
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! EIF signature production for the `EifSectionSignature` (0x04) section.
+//!
+//! [`LocalSigner`] signs in-process from a PEM ECDSA key.
+//! [`KmsSigner`] delegates to AWS KMS. Both emit the COSE raw ECDSA
+//! encoding `I2OSP(R, n) | I2OSP(S, n)` with `n = ceil(key_length / 8)`
+//! (RFC 9053 §2.1, which along with RFC 9052 obsoletes RFC 8152);
```
Comment (ginglis13) [inline tools/eif-builder/src/signer.rs]: Please just link to the RFC, like
```suggestion
//! (RFC 9053 §2.1: https://www.rfc-editor.org/rfc/rfc9053#section-2.1);
```

File: tools/eif-builder/src/signer.rs [hunk 11]
```diff
@@ -0,0 +1,668 @@
...
+//! P-256 → ES256, P-384 → ES384; other curves rejected. The TBS bytes
+//! passed to [`Signer::sign_cose`] are the COSE `Sig_structure1`
+//! serialization defined in RFC 9052 §4.4, built by the caller.
```
Comment (ginglis13) [inline tools/eif-builder/src/signer.rs]: Please link to the RFC

File: tools/eif-builder/src/signer.rs [hunk 12]
```diff
@@ -0,0 +1,668 @@
...
+            Self::Es384 => 48,
+        }
+    }
+}
+
+/// Backend for producing the COSE_Sign1 signature over the pre-hash TBS bytes.
+///
+/// `sign_cose` is `async` so KMS-backed implementations can await the SDK
+/// directly instead of blocking on an owned runtime. In-process backends do
+/// only CPU work and complete without yielding.
```
Comment (ginglis13) [inline tools/eif-builder/src/signer.rs]:
```suggestion
```

File: tools/eif-builder/src/lib.rs [hunk 13]
```diff
@@ -286,26 +320,539 @@ pub fn build_eif(
...
+            continue; // drop the old signature
+        }
+        entries.push(SectionEntry {
+            ty: s.ty,
+            data: Cow::Borrowed(&s.data),
+        });
+    }
+    // Check MAX_NUM_SECTIONS here so the error attributes overflow to resign.
+    ensure!(
+        entries.len() < MAX_NUM_SECTIONS,
```
Comment (sam-berning) [inline tools/eif-builder/src/lib.rs:718]: nit: I think this is slightly more readable:


```suggestion
        entries.len() <= MAX_NUM_SECTIONS - 1,
```

IMO, it just makes it clearer that we intentionally dropped one section.

**Review by sam-berning — APPROVED** [review]

**Review by ginglis13 — APPROVED** [review]
